src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunSymbolNodeFactory.cs

```diff
@@ -808,5 +815,40 @@ public override int GetHashCode()
             }
             return node;
         }
+
+        Dictionary<int, ObjectNode> _rvaFieldSymbols = new Dictionary<int, ObjectNode>();
+
+        public ObjectNode GetRvaFieldNode(FieldDesc fieldDesc)
+        {
+            Debug.Assert(fieldDesc.HasRva);
+            EcmaField ecmaField = (EcmaField)fieldDesc;
+            int rva = ecmaField.MetadataReader.GetFieldDefinition(ecmaField.Handle).GetRelativeVirtualAddress();
+            ObjectNode rvaFieldNode;
+            if (!_rvaFieldSymbols.TryGetValue(rva, out rvaFieldNode))
```

Comment:
I assume this is not going to do the right thing if the referenced field is in a different assembly from the one we're currently compiling. It's probably fine (such thing can only be constructed in IL), but it might be better to make it fail the compilation instead of generating bad code.

Reply:
Good point; I think I should change the dictionary to use the FieldDesc as the key; once we start doing "single file" i.e. compiling multiple assemblies into the same executable, this will become important.

Reply:
I have added two extra checks to the routine in 4th commit:

1) ThrowNotImpl when trying to access a cross-bubble RVA field (corresponds to an equivalent "TODO" in Crossgen);

2) ThrowNotImpl when compiling multiple input modules - this will need resolving when we start supporting "merged compilation", either by somehow referencing the RVA fields in the input MSIL modules or by copying the RVA field content to the R2R executable akin to what Crossgen does. We'll probably need some form of the latter technology if / once we start leveraging R2R for producing single-file outputs.